src/app/custom-editor/custom-editor.component.ts

```diff
@@ -172,12 +178,38 @@ export class CustomEditorComponent implements OnInit, OnDestroy, OnChanges {
     this.setValue(value);
     this.editorContentCopy = value;
     this.rawFile = value;
+    this.setContentType();
+    this.setAvailableViews();
     if (value != null || value !== '') {
       this.checkIfTextIsPretty();
     }
     if (this.showPrettifyOnLoad) {
-      this.onViewChange('xml');
+      if (this.contentType == 'xml') {
+        this.onViewChange('xml');
+        return;
+      }
+      if (this.contentType == 'json') {
+        this.onViewChange('json');
+        return;
+      }
```

Comment:
Couldn't this be:
```suggestion
      if (this.contentType) {
        this.onViewChange(this.contentType);
        return;
      }
```
Or at least
```suggestion
      if (this.contentType === 'xml') {
        this.onViewChange('xml');
        return;
      }
      if (this.contentType === 'json') {
        this.onViewChange('json');
        return;
      }
```
Thinking of this as a switch, what happens when the value does not match any of the cases?

Reply:
I've added the strict equality operator. 
To answer your question:
> what happens when the value does not match any of the cases?

Then the editor content cannot be prettified and therefore changes the view to raw